### PR TITLE
task 7 completed - voting bar module created and added to ArticlePage…

### PR DESCRIPTION
**Pull request by mikef80**
…. Uses optimistic rendering to update vote count

### CODE REVIEW

**Review by popatre — CHANGES_REQUESTED** [review]
- Nice to see the votes are persistent between refreshes
- Nice handling of the votes using optimistic rendering. Makes for a much better UX to have some instant feedback

- [x] We need to provide some feedback that something has gone wrong i.e. if the Internet connection is turned off, or the server has failed to do the request, we need to let the user know. 
At the moment, there's no feedback - it just reset but from a users perspective, not quite sure if something has gone wrong or not

File: src/components/VotingBar.jsx [hunk 1]
```diff
@@ -11,22 +12,35 @@ const VotingBar = ({ votes, article_id }) => {
       setArticleVotes((currVotes) => {
         return currVotes - vote_count;
       });
+
+      setShowErr(true);
+      
+      setTimeout(() => {
```
Comment (ghost) [inline src/components/VotingBar.jsx]: This should not be here, the error message should only disappear if there isn't a problem, such as someone tries again and the connection is fine. Your error state doesn't only last for three seconds.

**Review by ghost — CHANGES_REQUESTED** [review]
Not an ideal way to handle feedback to a user here. Keeping the error true until a rerender forces it back to false would be preferable.

File: src/components/VotingBar.jsx (hunk 1 above)
Comment (mikef80) [reply src/components/VotingBar.jsx]: cool.  I've reworked the code and submitted a new PR 🤞🏻

**Review by popatre — APPROVED** [review]
Looks all good to go 👍

**Pull request merged**